### PR TITLE
Fix "RefreshIndicator.color didn't update at runtime"

### LINKED ISSUE #73555: RefreshIndicator color didn't update at runtime
Hi Team,

I'm working on RefreshIndicator widget. I have faced an issue while changing the color of RefreshIndicator at runtime.

## Steps to Reproduce

1. For your reference, I have attached the dart file below.

```
class RefreshIndicatorDemo extends StatefulWidget {
  @override
  _RefreshIndicatorDemoState createState() => _RefreshIndicatorDemoState();
}

class _RefreshIndicatorDemoState extends State<RefreshIndicatorDemo> {
  Color _color = Colors.blue;

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('RefreshIndicator Demo'),
      ),
      body: RefreshIndicator(
        color: _color,
        onRefresh: () async {
          await Future.delayed(Duration(seconds: 3));
        },
        child: Column(
          children: [
            Expanded(
              child: SingleChildScrollView(
                scrollDirection: Axis.vertical,
                physics: AlwaysScrollableScrollPhysics(),
                child: Container(
                  height: 900,
                  color: Colors.white,
                ),
              ),
            ),
            FlatButton(
              color: Colors.blue,
              onPressed: () {
                setState(() {
                  _color = Colors.red;
                });
              },
              child: Text('Change color'),
            )
          ],
        ),
      ),
    );
  }
}
```

2. Please watch the video I have attached below.

https://user-images.githubusercontent.com/60499432/103999591-19e76600-51c3-11eb-826a-c27bba8d0884.mp4

**Expected results:** When tapping on the FlatButton, RefreshIndicator color has to be changed.

**Actual results:** When tapping on the FlatButton, RefreshIndicator color did not change.

<details>
  <summary>Logs</summary>
  
```
[ +108 ms] executing: [C:\src\flutter/] git -c log.showSignature=false log -n 1 --pretty=format:%H
[ +136 ms] Exit code 0 from: git -c log.showSignature=false log -n 1 --pretty=format:%H
[   +1 ms] b0a22998593fc605c723dee8ff4d9315c32cfe2c
[   +3 ms] executing: [C:\src\flutter/] git tag --points-at b0a22998593fc605c723dee8ff4d9315c32cfe2c
[ +139 ms] Exit code 0 from: git tag --points-at b0a22998593fc605c723dee8ff4d9315c32cfe2c
[   +1 ms] 1.25.0-8.2.pre
[  +93 ms] executing: [C:\src\flutter/] git rev-parse --abbrev-ref --symbolic @{u}
[  +74 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic @{u}
[        ] origin/beta
[        ] executing: [C:\src\flutter/] git ls-remote --get-url origin
[  +63 ms] Exit code 0 from: git ls-remote --get-url origin
[        ] https://github.com/flutter/flutter.git
[ +126 ms] executing: [C:\src\flutter/] git rev-parse --abbrev-ref HEAD
[  +74 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
[   +1 ms] beta
[ +134 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[   +1 ms] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[   +1 ms] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[   +6 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[   +1 ms] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[  +72 ms] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[  +51 ms] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[  +32 ms] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[   +1 ms] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[ +174 ms] executing: C:\Users\AshokKuvaraja\AppData\Local\Android\sdk\platform-tools\adb.exe devices -l
[  +96 ms] List of devices attached
           emulator-5554          device product:sdk_gphone_x86_arm model:AOSP_on_IA_Emulator device:generic_x86_arm
           transport_id:1
[  +21 ms] C:\Users\AshokKuvaraja\AppData\Local\Android\sdk\platform-tools\adb.exe -s emulator-5554 shell getprop
[ +260 ms] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[  +20 ms] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[  +65 ms] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[  +65 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[  +66 ms] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[  +28 ms] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[  +22 ms] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[   +3 ms] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[  +26 ms] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[   +2 ms] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[ +102 ms] "flutter run" took 1,140ms.
[  +40 ms] Target file "lib\main.dart" not found.
[   +4 ms] 
           #0      FlutterCommand.validateCommand (package:flutter_tools/src/runner/flutter_command.dart:1264:9)
           #1      RunCommand.validateCommand (package:flutter_tools/src/commands/run.dart:430:19)
           #2      FlutterCommand.verifyThenRunCommand (package:flutter_tools/src/runner/flutter_command.dart:1091:11)
           <asynchronous suspension>
           #3      FlutterCommand.run.<anonymous closure> (package:flutter_tools/src/runner/flutter_command.dart:986:27)
           <asynchronous suspension>
           #4      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #5      AppContext.run (package:flutter_tools/src/base/context.dart:149:12)
           <asynchronous suspension>
           #6      CommandRunner.runCommand (package:args/command_runner.dart:197:13)
           <asynchronous suspension>
           #7      FlutterCommandRunner.runCommand.<anonymous closure>
(package:flutter_tools/src/runner/flutter_command_runner.dart:274:9)
           <asynchronous suspension>
           #8      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #9      AppContext.run (package:flutter_tools/src/base/context.dart:149:12)
           <asynchronous suspension>
           #10     FlutterCommandRunner.runCommand (package:flutter_tools/src/runner/flutter_command_runner.dart:230:5)
           <asynchronous suspension>
           #11     run.<anonymous closure>.<anonymous closure> (package:flutter_tools/runner.dart:63:9)
           <asynchronous suspension>
           #12     run.<anonymous closure> (package:flutter_tools/runner.dart:61:12)
           <asynchronous suspension>
           #13     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #14     AppContext.run (package:flutter_tools/src/base/context.dart:149:12)
           <asynchronous suspension>
           #15     runInContext (package:flutter_tools/src/context_runner.dart:70:10)
           #16     main (package:flutter_tools/executable.dart:88:3)
           <asynchronous suspension>


[ +550 ms] ensureAnalyticsSent: 322ms
[  +69 ms] Running shutdown hooks
[   +1 ms] Shutdown hooks complete
[  +15 ms] exiting with code 1
```

```
[√] Flutter (Channel beta, 1.25.0-8.2.pre, on Microsoft Windows [Version 10.0.18363.418], locale en-US)
    • Flutter version 1.25.0-8.2.pre at C:\src\flutter
    • Framework revision b0a2299859 (3 days ago), 2021-01-05 12:34:13 -0800
    • Engine revision 92ae191c17
    • Dart version 2.12.0 (build 2.12.0-133.2.beta)

[√] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at C:\Users\AshokKuvaraja\AppData\Local\Android\sdk
    • Platform android-30, build-tools 30.0.3
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1248-b01)
    • All Android licenses accepted.

[√] Android Studio (version 3.3)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin version 34.0.1
    • Dart plugin version 182.5215
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1248-b01)

[√] VS Code, 64-bit edition (version 1.52.1)
    • VS Code at C:\Program Files\Microsoft VS Code
    • Flutter extension version 3.18.1

[√] Connected device (1 available)
    • AOSP on IA Emulator (mobile) • emulator-5554 • android-x86 • Android 9 (API 28) (emulator)

• No issues found!
```

</details>

### PR DESCRIPTION
**Pull request by pedromassango**
*List which issues are fixed by this PR. You must list at least one issue.*
Fixes https://github.com/flutter/flutter/issues/73555

### Tests
 - [x] Verify that `RefreshIndicator.color` can be updated at runtime

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

### CODE REVIEW

File: packages/flutter/lib/src/material/refresh_indicator.dart [hunk 1]
```diff
@@ -238,6 +238,22 @@ class RefreshIndicatorState extends State<RefreshIndicator> with TickerProviderS
...
+      final ThemeData theme = Theme.of(context);
+      _valueColor = _positionController.drive(
+        ColorTween(
+          begin: (widget.color ?? theme.accentColor).withOpacity(0.0),
+          end: (widget.color ?? theme.accentColor).withOpacity(1.0),
+        ).chain(CurveTween(
+            curve: const Interval(0.0, 1.0 / _kDragSizeFactorLimit)
+        )),
+      );
+    }
```
Comment (darrenaustin) [inline packages/flutter/lib/src/material/refresh_indicator.dart:254]: We might think about refactoring this out from here and `didChanceDependencies` to reduce some duplication, but it it is not critical.

**Review by darrenaustin — APPROVED** [review]
LGTM. Thx.

File: packages/flutter/lib/src/material/refresh_indicator.dart (hunk 1 above)
Comment (pedromassango) [reply packages/flutter/lib/src/material/refresh_indicator.dart:254]: I really thought about that, will send a fix today.

Comment (darrenaustin) [reply packages/flutter/lib/src/material/refresh_indicator.dart:254]: Given that this is already in, I wouldn't worry about it for this one. If we touch the file again we can take another look.

**Pull request merged**